examples/medplum-chart-demo/src/components/EncounterDetails.tsx

```diff
@@ -0,0 +1,110 @@
+import { Tabs } from '@mantine/core';
+import { getQuestionnaireAnswers, getReferenceString } from '@medplum/core';
+import { Encounter, Patient, Questionnaire, QuestionnaireResponse, Reference } from '@medplum/fhirtypes';
+import { Document, Loading, QuestionnaireForm, ResourceHistoryTable, ResourceTable, useMedplum } from '@medplum/react';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { EncounterNoteDisplay } from './EncounterNoteDisplay';
+
+interface EncounterDetailsProps {
+  encounter: Encounter;
+}
+
+export function EncounterDetails(props: EncounterDetailsProps): JSX.Element {
+  const medplum = useMedplum();
+  const navigate = useNavigate();
+  const [response, setResponse] = useState<QuestionnaireResponse>();
+  const [questionnaire, setQuestionnaire] = useState<Questionnaire>();
+
+  const id = props.encounter.id;
+
+  const tabs = ['Note', 'Details', 'History'];
+  const tab = window.location.pathname.split('/').pop();
+  const currentTab = tab && tabs.map((t) => t.toLowerCase()).includes(tab) ? tab : tabs[0].toLowerCase();
+
+  // Get the encounter type so the correct questionnaire can be retrieved
+  const code = props.encounter.type?.[0].coding?.[0].code;
+
+  useEffect(() => {
+    // Search for a response if there is one
+    medplum
+      .searchOne('QuestionnaireResponse', {
+        encounter: getReferenceString(props.encounter),
+      })
+      .then(setResponse)
+      .catch(console.error);
+
+    // Get the questionnaire
+    medplum
+      .searchOne('Questionnaire', {
+        // If the code is for gynecology or obstetrics, use it, otherwise search for the default
+        context: code === '163497009' || code === '83607001' ? code : '1287706006',
```

Comment:
make this more generic" 
```ts
code ?? GENERAL_ENCOUNTER_CODE
```

This way, the dev doesn't have to be aware of which specific codes are supported, and doesn't need to edit this line when adding a new one